code/game/machinery/atmos_exporter.dm

```diff
@@ -0,0 +1,262 @@
+					price = p_price
+			src.updateUsrDialog()
+		if ("scan")
+			for(var/obj/machinery/portable_atmospherics/canister/C in get_turf(src))
+				src.visible_message("<span class='danger'> Canister contains [LAZYLEN(C.air_contents.gases)] gases at [C.air_contents.return_pressure()]kpa pressure and with an internal temperature of [C.air_contents.temperature] kelvin.</span>")
+		if ("send")
+			for(var/obj/machinery/portable_atmospherics/canister/C in get_turf(src))
+				if((lpressure) <= C.air_contents.return_pressure() && C.air_contents.return_pressure() <= (upressure))
+					if((ltemp) <= C.air_contents.temperature && C.air_contents.temperature <= (utemp))
+						var/total_moles = C.air_contents.total_moles()
+						var/list/can_gas = C.air_contents.gases
+						var/i = 0
+						for(var/id in gas)
+							visible_message("<span class='danger'>NOW CHECKING [id], [can_gas[id]]</span>")
+							visible_message("<span class='danger'>[gas.len] versus [can_gas.len]</span>")
+							if(can_gas[id] && gas.len == can_gas.len)
+								var/gas_concentration = (can_gas[id][MOLES]/total_moles)*100
+								i++
+								visible_message("<span class='danger'>[gas_concentration] versus [concentration[i]]</span>")
+								if(gas_concentration <= (concentration[i]-5) || gas_concentration >= (concentration[i]+5))
+									visible_message("<span class='danger'>Invalid gas composition detected!</span>")
+									return
+							else
+								visible_message("<span class='danger'>Invalid gas detected!</span>")
+								return
+							visible_message("<span class='danger'>Valid [export_target] canister detected... exporting now. Your department will receive [price] points!</span>")
+							icon_state = "floorflush_a"
+							playsound(src, 'sound/machines/Ding.ogg', 100, 1)
+							spawn(20)
+								qdel(C)
+								icon_state = "floorflush_a2"
+								reset()
+					else
+						visible_message("<span class='danger'>Invalid temperature detected!</span>")
+				else
+					visible_message("<span class='danger'>Invalid pressure detected!</span>")
+			src.updateUsrDialog()
+		if ("close")
+			usr.unset_machine()
+
+/obj/machinery/atmos_points/proc/reset()
+	export_target = "------"
+	gas.Cut()
+	concentration.Cut()
+	upressure = 0
+	lpressure = 0
+	utemp = 0
+	ltemp = 0
+
+/obj/machinery/atmos_points/process()
+	if(market_cooldown < world.time)
+		economy_name = corp_name()
+		var/e_choice = rand(1,5)
+		switch(e_choice)
+			if(1)
+				e_gas += "o2"
+			if(2)
+				e_gas += "n2"
+			if(3)
+				e_gas += "n2o"
+			if(4)
+				e_gas += "plasma"
+			if(5)
+				e_gas += "co2"
+		e_pressure = rand(1,4500)
+		e_temp = rand(75,550)
+		e_concentration1 = 100
+		e_price = rand(200,500)
+
+		standard_name = corp_name()
+		for(var/i in 1 to 2)
+			var/s_choice = rand(1,5)
+			switch(s_choice)
+				if(1)
+					s_gas += "o2"
+				if(2)
+					s_gas += "n2"
+				if(3)
+					s_gas += "n2o"
+				if(4)
+					s_gas += "plasma"
+				if(5)
+					s_gas += "co2"

```

Comment:
Ill consider adding them in the future but putting them in this list would make atmos too dependent on cargo to fill orders.